src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.cs

```diff
@@ -232,6 +234,124 @@ public JsonNode Root
             }
         }
 
+        /// <summary>
+        /// Creates a new instance of the <see cref="JsonNode"/>. All children nodes are recursively cloned.
+        /// </summary>
+        public JsonNode DeepClone()
+        {
+            if (this is JsonObject jObject)
+            {
+                return jObject.DeepCloneObject();
+            }
+            else if (this is JsonArray jArray)
+            {
+                return jArray.DeepCloneArray();
+            }
+            else
+            {
+                return AsValue().DeepCloneValue();
+            }
+        }
+
+        /// <summary>
+        /// Returns <see cref="JsonValueKind"/> of current instance.
+        /// </summary>
+        public JsonValueKind GetValueKind()
+        {
+            if (this is JsonObject)
+            {
+                return JsonValueKind.Object;
+            }
+            else if (this is JsonArray)
+            {
+                return JsonValueKind.Array;
+            }
+            else
+            {
+                return AsValue().GetInternalValueKind();
+            }
```

Comment:
Can be simplified with a switch expression
```suggestion
            return this switch
            {
                JsonObject => JsonValueKind.Object.
                JsonArray => JsonValueKind.Array,
                _ => AsValue().GetInternalValueKind(),
            };
```